src/Field/Password.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+// vim:ts=4:sw=4:et:fdm=marker:fdl=0
+
+namespace atk4\data\Field;
+
+use atk4\core\InitializerTrait;
+
+class Password extends \atk4\data\Field
+{
+    use InitializerTrait {
+        init as _init;
+    }
+
+    public $type = 'password';
+
+    /**
+     * Keeping the actual hash protected, in case we have to validate password with
+     * compare().
+     *
+     * @var string
+     */
+    protected $password_hash = null;
+
+    public function init()
+    {
+        $this->_init();
+
+        $this->typecast = [
+            [$this, 'encrypt'],
+            function ($v, $f, $p) {
+                $this->password_hash = $v;
+                if ($p instanceof \atk4\ui\Persistence\UI) {
```

Comment:
I'm not particularly satisfied with this implementation, after all, Agile Data shouldn't have any references to Agile UI. 

Technically Password field implementation could have been moved into a separate add-on. Perhaps we should move it into `atk4/login` ?